### PR TITLE
feat: ResourceHierarchyLoader

### PR DESCRIPTION
**Pull request by sitepark-veltrup**
This pull request contains the ResourceHierarchyLoader with which resources can be loaded via its hierarchy links.

### CODE REVIEW

**Review by sitepark-schaeper — COMMENTED** [review]
All methods in `SiteKitResourceHierarchyLoader` and `SiteKitNavigationHierarchyLoader` that use the `ResourceLoader` can potentially throw `InvalidResourceException` and `ResourceNotFoundException`. I believe this is not specific to the sitekit-implementations and should probably already be denoted on the `ResourceLoader` and `ResourceHierarchyLoader` interfaces (and the sitekit implementations' methods respectively).

Oh, and the `composer.lock` seems to not be uptodate.

File: src/ResourceHierarchyLoader.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+interface ResourceHierarchyLoader
+{
+    public function loadRoot(string $location): Resource;
+
+    public function loadParent(string $location): ?Resource;
+
+    /**
+     * @return Resource[]
+     */
+    public function loadPath(string $location): array;
```
[the remark this answers is not in the document]
Comment (sitepark-veltrup) [reply src/ResourceHierarchyLoader.php:33]: I have completed the comment and described how the resources are arranged.

I do not yet see any need to return a special object here. From my point of view, it just makes it more complicated. In most cases I only have to iterate over the path. And that also works quite well with an array.

File: test/Loader/SiteKitLoaderTest.php [hunk 2]
```diff
@@ -22,6 +22,12 @@ protected function setUp(): void
         $this->loader = new SiteKitLoader($base);
     }
 
+    public function testExists(): void
+    {
+        $exists = $this->loader->exists('validResource.php');
+        $this->assertTrue($exists, 'resource should be exists');
```
[the remark this answers is not in the document]
Comment (sitepark-veltrup) [reply test/Loader/SiteKitLoaderTest.php]: adopted

File: src/ResourceHierarchyLoader.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (sitepark-schaeper) [reply src/ResourceHierarchyLoader.php:33]: Sure. If anything this is an easy addition at a later point.

**Review by sitepark-schaeper — APPROVED** [review]

**Pull request merged**